tests/e2e/test_torch_serialize.py

```diff
@@ -0,0 +1,72 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+import torch
+import random
+import string
+
+from azstoragetorch.io import BlobIO
+
+
+@pytest.fixture(scope="module", autouse=True)
+def torch_hub_cache(tmp_path_factory):
+    current_dir = torch.hub.get_dir()
+    torch.hub.set_dir(tmp_path_factory.mktemp("torch_hub"))
+    yield
+    torch.hub.set_dir(current_dir)
+
+
+@pytest.fixture(scope="module")
+def model():
+    model = torch.hub.load("pytorch/vision:v0.10.0", "resnet101", pretrained=False)
+    return model
+
+
+@pytest.fixture(scope="module")
+def model_path_name(tmp_path_factory):
+    return tmp_path_factory.mktemp("model") / f"{random_resource_name()}.pth"
+
+
+@pytest.fixture(scope="module", autouse=True)
+def upload_model(model, container_client, model_path_name):
+    torch.save(model.state_dict(), model_path_name)
+    blob_client = container_client.get_blob_client(blob=model_path_name.name)
+    with open(model_path_name, "rb") as f:
+        blob_client.upload_blob(f)
+
+
+@pytest.fixture()
+def state_dict_blob_url(account_url, container_client, model_path_name):
```

Comment:
I still think we may be leaving state between test cases because even though the `model_path_name` uses a random resource name, the fixture is set to `module` scope so it will be only chosen randomly once. Instead I'm thinking we:
1. Introduce a `container_url` fixture that returns the container url
2. Use the `container_url` fixture for the `state_dict_blob_url` fixture
3. For the `test_save_model` method, we use the `container_url` fixture instead of `state_dict_blob_url` and generate a new random resource name in the test body to guarantee we will be writing to a blob that did not previously exist nor be reused in other tests.